src/components/SwitchCase/SwitchCase.tsx

```diff
@@ -43,8 +52,65 @@ type Props<Case> = {
  *     />
  *   );
  * }
+ *
+ * @example
+ * type Ok<T> = { tag: "Ok"; value: T };
+ * type Err = { tag: "Err"; error: string };
+ * type Result<T> = Ok<T> | Err;
+ *
+ * function App() {
+ *   return (
+ *     <SwitchCase
+ *       value={result}
+ *       selector="tag"
+ *       caseBy={{
+ *         Ok: () => <ResultComponent data={result.value} />,
+ *         Err: () => <ErrorComponent />,
+ *       }}
+ *     />
+ *   );
+ * }
```

Comment:
What are the advantages of this approach compared to writing the code like below?

```suggestion
 * function App() {
 *   return (
 *     <SwitchCase
 *       value={result.tag}
 *       caseBy={{
 *         Ok: () => <ResultComponent data={result.value} />,
 *         Err: () => <ErrorComponent />,
 *       }}
 *     />
 *   );
 * }
```

Reply:
I Agree. It looks doesn't have any advantages for now 😂